# Retail Profit Recovery Analysis

## Notebook 1 – Data Understanding

### Project Overview

This project aims to investigate why a global retail company generates strong sales but does not achieve proportional profit growth.

Before performing any business analysis, it is essential to understand the available data, evaluate its quality, and identify the information that can support business decision-making.

This notebook focuses on understanding the dataset and preparing it for further analysis.

## Business Background

RetailX is a global retail company operating across multiple countries and markets.

Although revenue has grown consistently, management has observed that overall profitability has not improved at the same rate. Some products, customers, and regions appear to generate financial losses.

Management has requested a comprehensive analysis to identify the causes of low profitability and recommend strategies to improve business performance.

## Business Objective

The objectives of this project are to:

- Understand the structure and quality of the dataset.
- Prepare the data for analysis.
- Identify the key variables related to sales and profitability.
- Build a foundation for answering important business questions in later phases.

## Step 1: Import Required Libraries

The project begins by importing the Python libraries required for data analysis.

### Libraries Used

- **Pandas** – Data loading, cleaning, and analysis.
- **NumPy** – Numerical calculations and data manipulation.

These libraries form the foundation of most data analysis workflows.

In [1]:
import pandas as pd
import numpy as np

## Step 2: Load the Dataset

The Global Superstore dataset is loaded into a Pandas DataFrame.

The dataset contains transaction-level information including:

- Customer details
- Product information
- Sales
- Profit
- Discounts
- Shipping
- Geographic information

Each row represents an individual product purchased within an order.

In [2]:
df = pd.read_csv("superstore.csv")

### Step 3: Preview the Dataset

The first five records are displayed to gain an initial understanding of the dataset structure.

This helps verify that the data has been loaded correctly and provides an overview of the available variables.

In [6]:
df.head(0)


Empty DataFrame
Columns: [Category, City, Country, Customer.ID, Customer.Name, Discount, Market, 记录数, Order.Date, Order.ID, Order.Priority, Product.ID, Product.Name, Profit, Quantity, Region, Row.ID, Sales, Segment, Ship.Date, Ship.Mode, Shipping.Cost, State, Sub.Category, Year, Market2, weeknum]
Index: []

[0 rows x 27 columns]


### Observation

The dataset appears to contain one record per product sold within an order.

The available columns provide detailed information about customers, products, pricing, shipping, sales, and profit, making the dataset suitable for business analysis.

## Step 4: Understand Dataset Dimensions

Understanding the size of the dataset provides an estimate of the scale of analysis.

The number of rows represents individual transaction records, while the number of columns represents business attributes available for analysis.

In [8]:
df.shape

(51290, 27)

### Observation

The dataset contains **51290** transaction records and **27** variables.

The dataset is sufficiently large to identify meaningful business patterns and trends.

## Step 5: Explore Available Columns

Before performing analysis, it is important to understand the available business variables.

These columns represent different aspects of the business including customer information, products, sales performance, shipping operations, and geographical locations.

In [9]:
df.columns

Index(['Category', 'City', 'Country', 'Customer.ID', 'Customer.Name',
       'Discount', 'Market', '记录数', 'Order.Date', 'Order.ID', 'Order.Priority',
       'Product.ID', 'Product.Name', 'Profit', 'Quantity', 'Region', 'Row.ID',
       'Sales', 'Segment', 'Ship.Date', 'Ship.Mode', 'Shipping.Cost', 'State',
       'Sub.Category', 'Year', 'Market2', 'weeknum'],
      dtype='object')

### Observation

The dataset contains variables related to:

- Customer Information
- Product Information
- Sales Performance
- Profitability
- Discounts
- Shipping
- Geographic Markets

These variables will be used throughout the project to answer business questions.

## Step 6: Examine Data Types

Checking data types ensures that every column is stored in an appropriate format.

Correct data types are essential for accurate calculations, filtering, grouping, and time-based analysis.

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51290 entries, 0 to 51289
Data columns (total 27 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Category        51290 non-null  object 
 1   City            51290 non-null  object 
 2   Country         51290 non-null  object 
 3   Customer.ID     51290 non-null  object 
 4   Customer.Name   51290 non-null  object 
 5   Discount        51290 non-null  float64
 6   Market          51290 non-null  object 
 7   记录数             51290 non-null  int64  
 8   Order.Date      51290 non-null  object 
 9   Order.ID        51290 non-null  object 
 10  Order.Priority  51290 non-null  object 
 11  Product.ID      51290 non-null  object 
 12  Product.Name    51290 non-null  object 
 13  Profit          51290 non-null  float64
 14  Quantity        51290 non-null  int64  
 15  Region          51290 non-null  object 
 16  Row.ID          51290 non-null  int64  
 17  Sales           51290 non-null 

### Observation

The dataset contains both numerical and categorical variables.

The date columns will be reviewed during the data cleaning phase to ensure they are stored in the correct datetime format.

## Step 7: Generate Summary Statistics

Summary statistics provide a quick overview of the numerical variables.

This helps identify:

- Average values
- Minimum and maximum values
- Potential outliers
- Overall data distribution

In [11]:
df.describe()

,Discount,记录数,Profit,Quantity,Row.ID,Sales,Shipping.Cost,Year,weeknum
count,51290.000000,51290.0,51290.000000,51290.000000,51290.00000,51290.000000,51290.000000,51290.000000,51290.000000
mean,0.142908,1.0,28.610982,3.476545,25645.50000,246.498440,26.375818,2012.777208,31.287112
std,0.212280,0.0,174.340972,2.278766,14806.29199,487.567175,57.296810,1.098931,14.429795
min,0.000000,1.0,-6599.978000,1.000000,1.00000,0.000000,0.002000,2011.000000,1.000000
25%,0.000000,1.0,0.000000,2.000000,12823.25000,31.000000,2.610000,2012.000000,20.000000
50%,0.000000,1.0,9.240000,3.000000,25645.50000,85.000000,7.790000,2013.000000,33.000000
75%,0.200000,1.0,36.810000,5.000000,38467.75000,251.000000,24.450000,2014.000000,44.000000
max,0.850000,1.0,8399.976000,14.000000,51290.00000,22638.000000,933.570000,2014.000000,53.000000


### Observation

The summary statistics provide an overview of sales, profit, quantity, discount, and shipping cost.

These metrics will support future analysis of profitability and operational performance.

## Step 8: Identify Missing Values

Missing values can reduce data quality and affect business insights.

Before beginning analysis, it is important to identify whether any variables contain incomplete information.

In [13]:
df.isnull().sum()

,0
Category,0
City,0
Country,0
Customer.ID,0
Customer.Name,0
Discount,0
Market,0
记录数,0
Order.Date,0
Order.ID,0


### Observation

No missing values were identified in the dataset.

This indicates that the data is complete and suitable for further analysis.

## Step 9: Check for Duplicate Records

Duplicate records may distort business metrics such as total sales, profit, and customer counts.

The dataset is examined to determine whether duplicate transactions exist.

In [14]:
df.duplicated().sum()

np.int64(0)

Observation
No duplicate values were identified in the dataset.

This indicates that the data is complete and suitable for further analysis.

# Summary

In this notebook, the Global Superstore dataset was explored to understand its structure, quality, and business context.

### Key Findings

- The dataset contains detailed transaction-level sales information.
- Customer, product, sales, profit, shipping, and geographic data are available.
- The data quality assessment provides confidence for further analysis.
- The next notebook will focus on cleaning and preparing the data for business analysis.

With a clear understanding of the dataset, the project is now ready to move into the data cleaning phase.